cli/azd/pkg/auth/manager.go

```diff
@@ -422,12 +420,26 @@ func (m *Manager) newCredentialFromFederatedTokenProvider(
 		tenantID,
 		clientID,
 		func(ctx context.Context) (string, error) {
-			federatedToken, err := m.ghClient.TokenForAudience(ctx, "api://AzureADTokenExchange")
-			if err != nil {
-				return "", fmt.Errorf("fetching federated token: %w", err)
-			}
+			switch provider {
+			case gitHubFederatedAuth:
+				federatedToken, err := m.ghClient.TokenForAudience(ctx, "api://AzureADTokenExchange")
+				if err != nil {
+					return "", fmt.Errorf("fetching federated token: %w", err)
+				}
+
+				return federatedToken, nil
+			case gitLabFederatedAuth:
```

Comment:
I was really curious about what the "standard" is and looked around a little.
I think we can drop all the custom support and instead implement OIDC token support generically, following the footsteps of what's been done in other tools ([Pulumi](https://www.pulumi.com/blog/oidc-with-azure/#enabling-oidc), [Terraform](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_oidc#configuring-the-service-principal-in-terraform)) for better compatibility:

- `ACTIONS_ID_TOKEN_REQUEST_TOKEN` and `ACTIONS_ID_TOKEN_REQUEST_URL` are automatically detected for GitHub
- `ARM_OIDC_REQUEST_TOKEN` and `ARM_OIDC_REQUEST_TOKEN_URL` for override
- `ARM_OIDC_TOKEN` for direct token

Be happy to create an issue / make the changes here if you think it's worthwhile.

Reply:
Love it - feel free to take over the PR.